plugin/advisor/catalog/walk_through.go

```diff
@@ -116,11 +148,372 @@ func (d *databaseState) changeState(in tidbast.StmtNode) error {
+					table.comment = option.StrValue
+				case tidbast.TableOptionEngine:
+					table.engine = option.StrValue
+				}
+			}
+		case tidbast.AlterTableAddColumns:
+			for _, column := range spec.NewColumns {
+				pos := len(table.columnSet)
+				if spec.Position != nil && len(spec.NewColumns) == 1 {
+					var err error
+					pos, err = table.reorderColumn(spec.Position)
+					if err != nil {
+						return err
+					}
+				}
+				if err := table.createColumn(column, pos); err != nil {
+					return err
+				}
+			}
+			// MySQL can add table constraints in ALTER TABLE ADD COLUMN statements.
+			for _, constraint := range spec.NewConstraints {
+				if err := table.createConstraint(constraint); err != nil {
+					return err
+				}
+			}
+		case tidbast.AlterTableAddConstraint:
+			if err := table.createConstraint(spec.Constraint); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropColumn:
+			if err := table.dropColumn(spec.OldColumnName.Name.O); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropPrimaryKey:
+			if err := table.dropIndex(PrimaryKeyName); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropIndex:
+			if err := table.dropIndex(spec.Name); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropForeignKey:
+			// we do not deal with DROP FOREIGN KEY statements.
+		case tidbast.AlterTableModifyColumn:
+			if err := table.modifyColumn(spec.NewColumns[0], spec.Position); err != nil {
+				return err
+			}
+		case tidbast.AlterTableChangeColumn:
+			if err := table.changeColumn(spec.OldColumnName.Name.O, spec.NewColumns[0], spec.Position); err != nil {
+				return err
+			}
+		case tidbast.AlterTableRenameColumn:
+			if err := table.renameColumn(spec.OldColumnName.Name.O, spec.NewColumnName.Name.O); err != nil {
+				return err
+			}
+		case tidbast.AlterTableRenameTable:
+			if err := schema.renameTable(table.name, spec.NewTable.Name.O); err != nil {
+				return err
+			}
+		case tidbast.AlterTableAlterColumn:
+			if err := table.alterColumn(spec.NewColumns[0]); err != nil {
+				return err
+			}
+		case tidbast.AlterTableRenameIndex:
+			if err := table.renameIndex(spec.FromKey.O, spec.ToKey.O); err != nil {
+				return err
+			}
+		case tidbast.AlterTableIndexInvisible:
+			if err := table.changeIndexVisibility(spec.IndexName.O, spec.Visibility); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (t *tableState) changeIndexVisibility(indexName string, visibility tidbast.IndexVisibility) error {
+	index, exists := t.indexSet[indexName]
+	if !exists {
+		return NewIndexNotExistsError(t.name, indexName)
+	}
+	switch visibility {
+	case tidbast.IndexVisibilityVisible:
+		index.visible = true
+	case tidbast.IndexVisibilityInvisible:
+		index.visible = false
+	}
+	return nil
+}
+
+func (t *tableState) renameIndex(oldName string, newName string) error {
+	if strings.ToUpper(oldName) == PrimaryKeyName || strings.ToUpper(newName) == PrimaryKeyName {
```

Comment:
nit: add a comment to mention primary key name is not case sensitive if I understand the logic right.